### PR TITLE
Waiting meta-transactions

### PR DESCRIPTION
**Pull request by xitronix**

### CODE REVIEW

File: universal-login-relayer/lib/integration/sql/services/PendingMessagesSQLStore.ts [hunk 1]
```diff
@@ -66,7 +66,7 @@ export class PendingMessagesSQLStore implements IPendingMessagesStore {
     return {
       collectedSignatures: message.collectedSignatureKeyPairs.map((collected) => collected.signature),
       totalCollected: message.collectedSignatureKeyPairs.length,
-      required,
+      required: required.toNumber(),
```
Comment (marekkirejczyk) [inline universal-login-relayer/lib/integration/sql/services/PendingMessagesSQLStore.ts]: BigNumber

File: universal-login-sdk/lib/services/ExecutionFactory.ts [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+      return {} as MessageStatus;
+    };
+    return {
+      messageStatus: {} as MessageStatus,
+      waitForMined,
+      waitForPending
+    };
+  }
+
+  private isExecuted (messageStatus: MessageStatus){
```
Comment (marekkirejczyk) [inline universal-login-sdk/lib/services/ExecutionFactory.ts]: extract util

File: universal-login-sdk/lib/services/ExecutionFactory.ts [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+
+  }
+
+  async createExecution(signedMessage: SignedMessage): Promise<Execution> {
+    const result = await this.relayerApi.execute(stringifySignedMessageFields(signedMessage));
+    ensureNotNull(result.status.messageHash, MissingMessageHash);
+    const {messageHash, totalCollected, required} = result.status;
+    const waitForMined = totalCollected >= required ? this.createWaitForMined(messageHash) : async () => result.status;
+    const waitForPending = async () => {
+      return {} as MessageStatus;
```
Comment (marekkirejczyk) [inline universal-login-sdk/lib/services/ExecutionFactory.ts]: throw NotYetImplemented

File: universal-login-sdk/lib/services/ExecutionFactory.ts [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+    };
+    return {
+      messageStatus: {} as MessageStatus,
```
Comment (marekkirejczyk) [inline universal-login-sdk/lib/services/ExecutionFactory.ts]: implement messageStatus

File: universal-login-sdk/test/sdk.js [hunk 5]
```diff
@@ -67,59 +67,70 @@ describe('SDK - integration', async () => {
...
       it('should add a management key to the walletContract', async () => {
-        await sdk.addKey(contractAddress, otherWallet.address, privateKey, {gasToken: mockToken.address});
+        const {waitForMined} = await sdk.addKey(contractAddress, otherWallet.address, privateKey, {gasToken: mockToken.address});
+        await waitForMined();
         expect(await walletContract.getKeyPurpose(otherWallet.address)).to.be.eq(MANAGEMENT_KEY);
       });
 
       it('should add an action key to the walletContract', async () => {
-        await sdk.addKey(contractAddress, otherWallet.address, privateKey, {gasToken: mockToken.address}, ACTION_KEY);
+        const {waitForMined} = await sdk.addKey(contractAddress, otherWallet.address, privateKey, {gasToken: mockToken.address}, ACTION_KEY);
```
Comment (marekkirejczyk) [inline universal-login-sdk/test/sdk.js]: `waitForMined` -> `waitToBeMined()`
`waitForPending` -> `waitToBePending()`

File: universal-login-sdk/lib/services/ExecutionFactory.ts (hunk 3 above)
Comment (xitronix) [reply universal-login-sdk/lib/services/ExecutionFactory.ts]: ✔️

File: universal-login-sdk/lib/services/ExecutionFactory.ts (hunk 4 above)
Comment (xitronix) [reply universal-login-sdk/lib/services/ExecutionFactory.ts]: ✔️

File: universal-login-sdk/test/sdk.js (hunk 5 above)
Comment (xitronix) [reply universal-login-sdk/test/sdk.js]: ✔️

**Pull request merged**